text/0069-api-v2.md

```diff
@@ -185,7 +192,39 @@ Listed below is the compatibility matrix:
+
+There are two garbage collector existing in TiKV. One for transactional data, which utilizes [Compaction Filter] of RocksDB to drop obsolete versions during compaction, and schedules background tasks to physically remove deleted entires. Another one for RawKV data with TTL, which also utilizes [Compaction Filter] to drop TTL expired entries.
+
+`GC` of RawKV is quite similar with transactional data. First of all, check whether it's necessary to run `GC`:
+
+- For each [SST] in a compaction procedure, check [Table Properties] collected by customized collector. A [SST] will be `GC` when:
+  - The [SST] is at the bottommost level (as deleted & expired entries will only be recycled at bottommost level. See below). Or,
+  - There would be obsolete versions (`min_ts` *(minimum timestamp)* of [Table Properties] < `GC Safe Point`. Remind that newest version of a key is **NOT** an obsolete version), and,
+  - More than a ratio (defaults to `10%`) of all versions are obsolete.
+- Then, `GC` will be invoked when:
+  - The compaction is at bottommost level, or
+  - More than half of the [SST] files in the compaction procedure need `GC`, or
+  - Sum of [Table Properties] of all [SST] meets the same condition as for single [SST].
+
+Then, in [Compaction Filter]:
+
+- For entries with timestamp >= `GC Safe Point`, mark as `Keep`.
+- For entries with timestamp < `GC Safe Point` and **NOT** the newest one of a key, mark as `Remove`. Then RocksDB will filter them out during compaction.
+- For entries: 1. with timestamp < `GC Safe Point`, 2. they **ARE** the newest one of a key (in current [SST]. It's too expensive to check for the whole LSM tree), and 3. it's deleted or expired: schedules a background task to scan all versions older than it (of the same key), and remove all of them by issuing a `Delete` to RocksDB.
+
+More about `GC` of deleted/expired entries:
+
+- The reason for recycling by scan & delete other than mark as `Remove` is that: Filter just the newest one will cause user getting older versions, and lead to wrong result.
+- The reason for recycling only at bottommost level: Deletion is implemented by RocksDB as inserting `Tombstone` records at the top of LSM tree, other than remove them immediately. So if deletes the versions not at bottommost level, it's likely to see them more times in another compaction procedures later on, and issue more deletes. That would lead to not recycle but occupy more space, and more `GC` overhead.
```

Comment:
```suggestion
- The reason for recycling only at bottommost level: Deletion is implemented by RocksDB as inserting `Tombstone` records at the top of LSM tree, rather than remove them immediately. So if deletes the versions not at bottommost level, it's likely to see them more times in another compaction procedures later on, and issue more deletes. That would lead to not recycle but occupy more space, and more `GC` overhead.
```